{% extends "layout.html" %}

{% block title %}Engineering{% endblock %}

{% block extra_css %}
<link rel="stylesheet" href="{{ url_for('static', filename='main.css') }}">
<link rel="stylesheet" href="{{ url_for('static', filename='engineering.css') }}">
{% endblock %}

{% block page_content %}

<div id="dbtl-component">

<h1 class="dbtl-header">Wet Lab</h1>

<div class="cycle-card">
    <div class="cycle-left">
        <div>
            <h2>Cycle 1</h2>
            <div class="phase-tabs">
                <div class="phase-tab active" data-tab="design1">Design</div>
                <div class="phase-tab" data-tab="build1">Build</div>
                <div class="phase-tab" data-tab="test1">Test</div>
                <div class="phase-tab" data-tab="learn1">Learn</div>
            </div>
        </div>
    </div>
    <div class="cycle-right">
        <div class="phase-content active" id="design1">
            <div class="phase-title">Design</div>
            <div class="phase-subtitle">Establishment of metabolic pathway(s) and genes</div>
            <p>Via an extensive literature review, we identified the Dissimilatory Nitrate Reduction to Ammonium (DNRA) pathway and the denitrification pathways, along with the operons that control them.</p>

            <div class="phase-subtitle">Identification of bacteria</div>
            <p>We identified <em>Pseudomonas putida</em> Y-9, a bacterial strain that had been isolated from long-term flooded paddy fields in China. At the time, we believed this to be the only bacterial species in the world that could perform DNRA under aerobic conditions. <em>P. putida</em> Y-9 also performed denitrification under aerobic conditions.</p>

            <div class="phase-subtitle">Denitrification knockout</div>
            <p>The first part of the plan was to delete the <em>nirK/nirS</em> genes responsible for the metabolic conversion of nitrate to nitric oxide. Stopping the expression of this operon would ensure that denitrification does not proceed, and DNRA would hypothetically automatically be preferred by the bacterium.</p>

            <div class="phase-subtitle">DNRA upregulation</div>
            <p>Simultaneously, we also planned to upregulate the expression of DNRA. This would ensure that nitrate build-up did not increase up to and beyond its toxicity limit for the organism, while ensuring that ammonium was continuously churned out by the cell.</p>
        </div>
        <div class="phase-content" id="build1">
            <div class="phase-title">Build</div>
            <p>Our first step was to reach out to the researchers who had discovered this novel organism. However, we could not get a response, no matter how desperately we tried to contact them. The genomic data of the organism were also not available online because it had not been fully sequenced yet.</p>
        </div>
        <div class="phase-content" id="test1">
            <div class="phase-title">Test</div>
            <p>Due to our lack of resources, the testing phase in this cycle was geared toward testing the validity and feasibility of our idea.</p>
            <br>
            <div class="callout-box"><p>We therefore spoke to Prof Ramesh Chand, chairperson of NITI Aayog, and Shyam Sundar Agarwal, Director of the Ministry of Chemicals and Fertilizers. Both are major stakeholders in the field of bioremediation and were interested in our problem statement and approach to the solution. They reinforced our hypothesis that nitrate leaching was a real, imminent problem and that researchers across the board were looking for viable solutions.</p></div>
        </div>
        <div class="phase-content" id="learn1">
            <div class="phase-title">Learn</div>
            <div class="phase-subtitle">Characterization of gene dynamics:</div>
            <p>DNRA being extremely understudied meant that we did not know which promoter gene was controlling its activity, and therefore did not know how it would respond to various environmental conditions. We learnt how to operate the NIH Genbank to search for other bacteria, at least the genes that were mentioned by name in the papers by the Chinese researchers. We then scoured the ATCC (American Type Culture Collection) and MTCC to locate what new bacteria were available to us.</p>
            <br>
            <div class="callout-box"><p>While this was eventually a dead end, the unwieldy nature of the website inspired us to build a better tool with all the information within Genbank, but with a much better and user-friendly graphical user interface. We hope the tool will prove useful to biologists who may be as fed up with Genbank as we were.</p></div>
            <div class="phase-subtitle">Defining a knockout mechanism:</div>
            <p>Gene deletion has quite a few standard procedures, like homologous recombination and conditional knockout. Our professors advised us to identify the method that would best work for our skill level and laboratory equipment. We quickly learned that knockout might not be a feasible option for us, because it was unstable and might not work as expected when dealing with an entire metabolic pathway. We had to reassess.</p>
            <br>
            <div class="callout-box"><div class = "phase-subtitle">Building upon Team Cattlysts:</div>
            <p>Team Cattlelyst from Wageningen University had already demonstrated the first step of the pathway, nitrate reduction to nitrite, in <em>P. putida</em> KT2440 during their own iGEM cycle. This was huge because it meant that we could work on perfecting only the nitrite reduction to ammonium pathway and could use their results to establish our proof of concept.</p>
            </div>
        </div>
    </div>

    <div class="plasmid">
        <svg viewBox="0 0 100 100">
            <defs>
                <marker id="arrowhead" markerWidth="2.5" markerHeight="2.5" refX="1.25" refY="1.25" orient="auto">
                    <polygon points="0 0, 2.5 1.25, 0 2.5" fill="var(--phase-build)" />
                </marker>
            </defs>
        
            <circle class="plasmid-bg" cx="50" cy="50" r="40"/>

            <g transform="rotate(-90 50 50)">
                <path id="segment-1" class="plasmid-segment" d="M 53.49 10.3 A 40 40 0 0 1 89.7 46.51" />
                <path id="segment-2" class="plasmid-segment" d="M 90 53.49 A 40 40 0 0 1 53.49 89.7" marker-end="url(#arrowhead)" />
                <path id="segment-3" class="plasmid-segment" d="M 13.62 66.86 A 40 40 0 0 0 33.14 86.38" />
                <path id="segment-4" class="plasmid-segment" d="M 10 46.51 A 40 40 0 0 1 46.51 10.3" />
            </g>
        </svg>
    </div>
</div>


<div class="cycle-card">
    <div class="cycle-left">
        <div>
            <h2>Cycle 2</h2>
            <div class="phase-tabs">
                <div class="phase-tab active" data-tab="design2">Design</div>
                <div class="phase-tab" data-tab="build2">Build</div>
                <div class="phase-tab" data-tab="test2">Test</div>
                <div class="phase-tab" data-tab="learn2">Learn</div>
            </div>
        </div>
    </div>
        <div class="cycle-right">
            <div class="phase-content active" id="design2">
                <div class="phase-title">Design</div>
                <div class="phase-subtitle">Isolation of new aerobic DNRA bacteria</div>
                <p>After further digging, we stumbled upon an extremely recent paper that had isolated three more bacterial species that performed aerobic DNRA. <em>Bacillus salipaludis</em> PS3-36, <em>Neobacillus</em> sp. PS2-9, and PS3-12 had been isolated from another set of paddy fields, this time from Iksan city, Korea. These bacteria also performed DNRA under aerobic conditions, and were the first Bacillus species to carry out that pathway.</p>

                <div class="phase-subtitle">Next steps and challenges</div>
                <p>Our next steps remained the same, wherein we would downregulate denitrification within the bacterium and upregulate DNRA. This meant our issues were also identical, with the largest problem being that downregulation was still a bit of a black box in terms of how we would actually carry it out.</p>
            </div>

            <div class="phase-content" id="build2">
                <div class="phase-title">Build</div>
                <p>Since the genomes of these bacterial species were publicly available data, we rejoiced. After an appropriate amount of rejoicing, we got down to business and began to look into how we could obtain them.</p>
                <br>
                <p>We first tried and failed to get in touch with the researchers who had published the paper. Next, we scoured MTCC and ATCC to no avail. One of our secondary PIs, Professor Umesh Varshney, advised us to locate local genetic banks or microbial culture collections to see if they housed the strains.</p>
                <div class="callout-box">
                    <p>Thus, we spoke to the curator at the MACS Collection of Microorganisms (MCM) at the Agharkar Research Institute in Maharashtra, India. They could, unfortunately, not help us because they did not house the strains either. They did, however, contain <em>Bacillus subtilis</em>, a common nitrifying bacterium. Prof Varshney expressed that it would be a good idea if we could work with <em>Bacillus subtilis</em> since his laboratory had previously worked with them, and the established protocols would prove extremely helpful to us during experimentation.</p>
                </div>
            </div>

            <div class="phase-content" id="test2">
                <div class="phase-title">Test</div>
                <p>Since we could not test whether <em>Bacillus subtilis</em> contained DNRA genes in the wet lab just yet, we turned to publicly available data. Our dry lab team assessed GEM models, which are networks of all the metabolites, enzymes, and corresponding genes involved within a bacterial species. Since <em>Bacillus subtilis</em> had a pre-established GEM model, we ran experiments to check whether it could theoretically perform DNRA, and whether it was a good idea at all to even try and procure the strain. We did not obtain conclusive evidence through these trials.</p>
                <br>
                <p>Furthermore, we extensively used the BLAST tool from Genbank to compare the genes that carried out this aerobic DNRA pathway in the Bacillus and Neobacillus species with pre-established anaerobic DNRA carried out by other microbes. To our utter surprise, the proteins produced were identical.</p>

            </div>

            <div class="phase-content" id="learn2">
                <div class="phase-title">Learn</div>
                <div class="phase-subtitle">Breakthroughs</div>
                <br>
                <p><strong>The rise and fall of Bacillus:</strong></p>
                <p>While we were initially extremely excited about these novel bacteria and the fact that <em>Bacillus subtilis</em> was a regular in our professor’s laboratory, we quickly stumbled upon a major trade-off. Bacillus species are notorious for forming spores. Considering that our eventual plan was environmental deployment, this would present a major biosafety challenge. While dismantling the spore-forming properties of the bacteria via gene editing would be possible, it would create another layer of uncertainty that we were ill-equipped to deal with.</p>
                <br>
                <p><strong>The Other Genes:</strong></p>
                <p>While we had mostly been focused on <em>nirBD</em> as the genes responsible for nitrite reduction to ammonium, the <em>nrf</em> operon emerged during literature review. This operon existed as two variations across bacterial species: <em>nrfABCDEFG</em> and <em>nrfHAIJ</em>. For greater detail, please refer to the Design page. Moving forward, we have considered all three genes in our experimentation and further DBTL cycles.</p>
                <br>
                <p><strong>The breakthrough, and our solution to the aerobicity problem:</strong></p>
                <p>Our biggest hurdle so far has been the difficulty in obtaining these novel and exotic strains, something we describe in detail in our blog post. Researchers were unresponsive, and it seemed that the elusive property of these bacteria to perform DNRA aerobically would elude us forever. However, the tests we ran by BLASTing proteins and nucleotide sequences together gave us the connecting link needed to identify the research gap in this domain.</p>
                <br><br>
                <div class="callout-box">
                    <p>1. We could decouple the operon from its promoter gene and attach it to a constitutive promoter of our choice instead. This would allow non-DNRA performing organisms, or organisms that only perform DNRA in anaerobic environments, to perform DNRA aerobically. We could create our own novel bacterium and tweak it to work under our own conditions. These revelations are largely credited to former iGEM participant and our senior, Aditya Kamath Ammembal, who helped us connect these dots.</p>
                </div>
                <div class="callout-box">
                    <p>2. We could disregard denitrification entirely. If we were to engineer non-DNRA bacteria and introduce this metabolic pathway into an ecosystem that previously did not contain it, along with the assurance that the proteins were stable enough to carry out their function, we would not need to suppress denitrification to reduce nitrogen loss from the soil.</p>
                </div>
                <div class="callout-box">
                    <p>3. In this period, we spoke to Dr. Priyanka Jamwal, researcher at ATREE (Ashoka Trust For Research In Ecology And The Environment). She agreed with the severity of the problem and directed us to groundwater pollution, an avenue we hadn’t considered yet. Her input on soak pits, source pollution, and rural groundwater dependence provided great context within which we grounded our project moving forward.</p>
                </div>
            </div>
        </div>

    <div class="plasmid">
        <svg viewBox="0 0 100 100">
            <defs>
                <marker id="arrowhead" markerWidth="2.5" markerHeight="2.5" refX="1.25" refY="1.25" orient="auto">
                    <polygon points="0 0, 2.5 1.25, 0 2.5" fill="var(--phase-build)" />
                </marker>
            </defs>
        
            <circle class="plasmid-bg" cx="50" cy="50" r="40"/>

            <g transform="rotate(-90 50 50)">
                <path id="segment-1" class="plasmid-segment" d="M 53.49 10.3 A 40 40 0 0 1 89.7 46.51" />
                <path id="segment-2" class="plasmid-segment" d="M 90 53.49 A 40 40 0 0 1 53.49 89.7" marker-end="url(#arrowhead)" />
                <path id="segment-3" class="plasmid-segment" d="M 13.62 66.86 A 40 40 0 0 0 33.14 86.38" />
                <path id="segment-4" class="plasmid-segment" d="M 10 46.51 A 40 40 0 0 1 46.51 10.3" />
            </g>
        </svg>
    </div>
</div>

<div class="cycle-card">
    <div class="cycle-left">
        <div>
            <h2>Cycle 3</h2>
            <div class="phase-tabs">
                <div class="phase-tab active" data-tab="design3">Design</div>
                <div class="phase-tab" data-tab="build3">Build</div>
                <div class="phase-tab" data-tab="test3">Test</div>
                <div class="phase-tab" data-tab="learn3">Learn</div>
            </div>
        </div>
    </div>
    <div class="cycle-right">
        <div class="phase-content active" id="design3">
            <div class="phase-title">Design</div>
            <p>We decided to introduce the genes into a non-DNRA-performing bacterium via a plasmid. With our breakthrough, we could now bypass the aerobicity problem and also use a much more easily accessible bacterial species. This bacterium of choice turned out to be <em>Pseudomonas putida</em> KT2440, a microbial workhorse and robust soil bacterium.</p>

            <div class="phase-subtitle">Choice of chassis</div>
            <p>One of the reasons we picked this chassis was because a previous team at our institute had also used <em>P. putida</em> KT2440 for their project, and had received it from Prof Prashant Phale at the Indian Institute of Technology, Bombay. We reached out to him, and his lab graciously sent us the agar stab containing the bacterium.</p>

            <div class="phase-subtitle">Plasmid selection</div>
            <p>We decided to use plasmids from the SEVA repository because they were free and we were working with an extremely tight budget. Furthermore, the SEVA plasmids were well-characterized and easy to work with due to easy online access and sourcing. We also sourced <em>P. putida</em> EM42 from SEVA, which is a strain of KT2440 with over 200 inessential genes knocked out. This strain might therefore be more likely to take up external genes due to the lower level of intrinsic metabolic burden.</p>
        </div>

        <div class="phase-content" id="build3">
            <div class="phase-title">Build</div>
            <p>We digitally modified our operons and added restriction enzymes to the ends for blunt-end restriction enzyme digestion. We were to receive the operons in clonal vectors from TWIST Bioscience, then digest them and ligate them into pSEVA expression vectors. We built digital maps of all our plasmids. All of the genes and vectors have been uploaded as BioBricks.</p>

            <div class="callout-box">
                <strong>Plasmid maps:</strong> 
                <div class="accordion-item">
                        <button class="accordion-header">
                            <span><em>nrfABCDEFG</em> in pSEVA648</span>
                            <span class="accordion-icon">+</span>
                        </button>
                        <div class="accordion-content">
                        <img src="https://static.igem.wiki/teams/6006/gallery/wetlab/nrfabcdefg.avif" alt="Plasmid maps" class="animate-on-scroll" style="display: block; margin: 1rem auto 0; max-width: 450px; border-radius: 1rem; box-shadow: 0 10px 15px -3px rgba(0,0,0,0.1); margin-bottom: 2rem;"> 
                        </div>
                    </div>
                <div class="accordion-item">
                        <button class="accordion-header">
                            <span><em>nirBD</em> in pSEVA648</span>
                            <span class="accordion-icon">+</span>
                        </button>
                        <div class="accordion-content">
                        <img src="https://static.igem.wiki/teams/6006/gallery/wetlab/pseva648-nirbd.avif" alt="Plasmid maps" class="animate-on-scroll" style="display: block; margin: 1rem auto 0; max-width: 450px; border-radius: 1rem; box-shadow: 0 10px 15px -3px rgba(0,0,0,0.1); margin-bottom: 2rem;"> 
                        </div>
                    </div>
                <div class="accordion-item">
                        <button class="accordion-header">
                            <span><em>pSEVA648 map</em></span>
                            <span class="accordion-icon">+</span>
                        </button>
                        <div class="accordion-content">
                        <img src="https://static.igem.wiki/teams/6006/gallery/wetlab/pseva648.avif" alt="Plasmid maps" class="animate-on-scroll" style="display: block; margin: 1rem auto 0; max-width: 450px; border-radius: 1rem; box-shadow: 0 10px 15px -3px rgba(0,0,0,0.1); margin-bottom: 2rem;"> 
                        </div>
                    </div>
                <div class="accordion-item">
                        <button class="accordion-header">
                            <span><em>pSEVA258 map</em></span>
                            <span class="accordion-icon">+</span>
                        </button>
                        <div class="accordion-content">
                        <img src="https://static.igem.wiki/teams/6006/gallery/wetlab/pseva258.avif" alt="Plasmid maps" class="animate-on-scroll" style="display: block; margin: 1rem auto 0; max-width: 450px; border-radius: 1rem; box-shadow: 0 10px 15px -3px rgba(0,0,0,0.1); margin-bottom: 2rem;"> 
                        </div>
                    </div>
                <div class="accordion-item">
                        <button class="accordion-header">
                            <span><em>pSEVA238 map</em></span>
                            <span class="accordion-icon">+</span>
                        </button>
                        <div class="accordion-content">
                        <img src="https://static.igem.wiki/teams/6006/gallery/wetlab/pseva238.avif" alt="Plasmid maps" class="animate-on-scroll" style="display: block; margin: 1rem auto 0; max-width: 450px; border-radius: 1rem; box-shadow: 0 10px 15px -3px rgba(0,0,0,0.1); margin-bottom: 2rem;"> 
                        </div>
                    </div>
                
            </div>
        </div>

        <div class="phase-content" id="test3">
            <div class="phase-title">Test</div>
            <p>When we cross-checked the amino acid sequences of our newly-built plasmids against the original protein sequences on Genbank, we found perfect matches. We were assured that our genes should be perfectly expressed. This enabled us to order the genes from TWIST Bioscience and prepare for the cycles ahead.</p>
        </div>

        <div class="phase-content" id="learn3">
            <div class="phase-title">Learn</div>
            <p>While this was a technical and highly specific issue, our operons were simply too large for TWIST Bioscience to synthesize in their entirety in the same plasmid. Therefore, we needed to split each of our operons so that they could fit within the synthesizable 5kbp limit.</p>

            <div class="phase-subtitle">Proof-of-concept in <em>E. coli</em></div>
            <p>Professors in the Microbiology and Cell Biology Department at our institute didn’t really work with <em>P. putida</em> KT2440. Therefore, we were advised to demonstrate our proof-of-concept in <em>Escherichia coli</em> DH5-Alpha, which would be easier to work with in our context.</p>
            <div class="callout-box">
                <p>Ramesh Vaidyanathan, a former student of one of our advisors, Prof Mahavir Sing, and current Business Development Manager at TWIST Biosciences, graciously provided input into how we could separately optimize our operons for expression in both bacteria using the Codon Adaptation Index theory.</p>
            </div>
            <div class="phase-subtitle">Biofilm considerations & biosafety</div>
            <p>We soon learned that <em>P. putida</em> KT2440 forms biofilms. Although biofilms are generally notorious for causing disease, <em>P.putida</em> KT2440 is actually a class of organisms that forms Plant Growth Promoting Biofilms (PGPB). These biofilms confer a multitude of benefits to the plant roots, as detailed in the project overview.</p>
            <div class="callout-box"><p>While deciding upon pSEVA plasmids, we stumbled upon multiple vectors with inducible promoter systems. At the All-India iGEM Meet, iGEM ambassadors conducted an enlightening workshop on biosafety, which planted the seed in our search for implementing some type of mechanism to ensure biosafety for our engineered organism. The biofilm-forming properties of the organism meant that the bacteria would be adhering to the plant roots, which inspired us to link the expression of the operon to an inducible promoter with a plant root exudate as the inducer.</p></div>
            <div class="callout-box">
            <div class="phase-subtitle">SEVA plasmid challenges</div>
            <p>We tried to import the vector pSEVA 248 (kanamycin-resistant) multiple times from the SEVA repository, but received an unviable bacterium <em>(E. coli)</em> every time. After three attempts, we reached back out to Prof Phale, who had a tie-up with SEVA. His lab had previously worked with them, and they graciously received the bacterium to send to us. Although they too failed in culturing the 248 bacterium, they did send us other SEVA plasmids with the same Origin of Replication and promoter system (XylS/Pm): pSEVA 238, 258, 648 (kanamycin-resistant), and the constitutive promoter pSEVA 2313 (gentamicin-resistant).</p>
            </div>
        </div>
    </div>

    <div class="plasmid">
        <svg viewBox="0 0 100 100">
            <defs>
                <marker id="arrowhead" markerWidth="2.5" markerHeight="2.5" refX="1.25" refY="1.25" orient="auto">
                    <polygon points="0 0, 2.5 1.25, 0 2.5" fill="var(--phase-build)" />
                </marker>
            </defs>
        
            <circle class="plasmid-bg" cx="50" cy="50" r="40"/>

            <g transform="rotate(-90 50 50)">
                <path id="segment-1" class="plasmid-segment" d="M 53.49 10.3 A 40 40 0 0 1 89.7 46.51" />
                <path id="segment-2" class="plasmid-segment" d="M 90 53.49 A 40 40 0 0 1 53.49 89.7" marker-end="url(#arrowhead)" />
                <path id="segment-3" class="plasmid-segment" d="M 13.62 66.86 A 40 40 0 0 0 33.14 86.38" />
                <path id="segment-4" class="plasmid-segment" d="M 10 46.51 A 40 40 0 0 1 46.51 10.3" />
            </g>
        </svg>
    </div>
</div>

<div class="cycle-card">
    <div class="cycle-left">
        <div>
            <h2>Cycle 4</h2>
            <div class="phase-tabs">
                <div class="phase-tab active" data-tab="design4">Design</div>
                <div class="phase-tab" data-tab="build4">Build</div>
                <div class="phase-tab" data-tab="test4">Test</div>
                <div class="phase-tab" data-tab="learn4">Learn</div>
            </div>
        </div>
    </div>
    <div class="cycle-right">
        <div class="phase-content active" id="design4">
            <div class="phase-title">Design</div>
            <p>We now plan to use CAI to design the genes for optimum expression in both <em>E. coli</em> and <em>P. putida</em> separately. The operon would be working under an inducible promoter, and our inducer of choice was 2-methylbenzoate, a plant root exudate that is released in abundance by crops like maize and wheat.</p>

            <div class="phase-subtitle">Water bioremediation & biofilm framework</div>
            <p>Simultaneously, we ideated a comprehensive system where our engineered microbe could be used in water bioremediation, wherein the final product would be ammonium-rich water that could be used in lieu of fertilizers, thus increasing fertilizer efficiency even further. Furthermore, we built a framework for the possible ways in which we could actually inoculate plant roots with the engineered bacterium for maximum biofilm formation, adhesion, and DNRA efficiency in terms of proximity to both 2-methylbenzoate and nitrate.</p>
        </div>

        <div class="phase-content" id="build4">
            <div class="phase-title">Build</div>
            <p>We built the operons, courtesy of our sponsor Snapgene, and digitally inserted them into pSEVA vectors 238, 258, 648, and 2313. Having cross-checked amino acid sequences, we placed the order and received our operons from TWIST Bioscience.</p>
            <br>
            <p>TWIST Bioscience was not able to synthesize one-half of our operons due to manufacturing issues, meaning we now had two operons to work with instead of three. Therefore, we compiled protocols to isolate, digest, and ligate the operons with the pSEVA vectors before transforming <em>E.coli</em> DH5-Alpha.</p>
            <br>
            <p>We also built nitrite and ammonium testing assays for confirming DNRA in the transformed bacterium, and biofilm assays for confirming basal level and post-transformation biofilm-forming capabilities of <em>P. putida</em>.</p>
        </div>

        <div class="phase-content" id="test4">
            <div class="phase-title">Test</div>
            <p>We isolated the plasmids from the shipping bacteria and digested them with our designed enzymes, then checked the DNA concentrations via a nanodrop machine. This process contained multiple DBTL cycles within itself, but we managed to obtain working concentrations for all DNA fragments multiple times. Find the details of how we troubleshooted issues during this phase in the Experimentation page.</p>
            <br>
            <p>We performed DNA ligation to connect the expression vector to the operons and transformed <em>E. coli</em>. Unfortunately, we did not see growth after our first transformation on Km agar plates. However, our dry lab modelled exceptional results, as detailed in the dry lab engineering webpage.</p>
            <br>
            <p>Our base-level tests for DNRA in <em>P.putida</em> started with optimizing the growth medium for <em>P. putida</em> KT2440 by testing different broth compositions from multiple sources. We then picked cultures with the highest growth and checked for base-levels of DNRA in <em>P. putida</em>, a test never before carried out with this specific bacterium, even though it does contain the nirBD genes that should cause production of nitrite reductase.</p>
        </div>

        <div class="phase-content" id="learn4">
            <div class="phase-title">Learn</div>
            <p>Plasmid kits can contaminate DNA samples with RNA and salts. To bypass this problem, we conducted manual plasmid isolations and compared the results as seen on the experimentation page.</p>
            <br>
            <p>Minimal growth media seemed to be the best for <em>P. putida</em> KT2440 growth, which was an interesting revelation considering literature consistently shows that a high C/N ratio is favored for DNRA. We discuss the implications on our experimentation page.</p>
        </div>
    </div>

    <div class="plasmid">
        <svg viewBox="0 0 100 100">
            <defs>
                <marker id="arrowhead" markerWidth="2.5" markerHeight="2.5" refX="1.25" refY="1.25" orient="auto">
                    <polygon points="0 0, 2.5 1.25, 0 2.5" fill="var(--phase-build)" />
                </marker>
            </defs>
        
            <circle class="plasmid-bg" cx="50" cy="50" r="40"/>

            <g transform="rotate(-90 50 50)">
                <path id="segment-1" class="plasmid-segment" d="M 53.49 10.3 A 40 40 0 0 1 89.7 46.51" />
                <path id="segment-2" class="plasmid-segment" d="M 90 53.49 A 40 40 0 0 1 53.49 89.7" marker-end="url(#arrowhead)" />
                <path id="segment-3" class="plasmid-segment" d="M 13.62 66.86 A 40 40 0 0 0 33.14 86.38" />
                <path id="segment-4" class="plasmid-segment" d="M 10 46.51 A 40 40 0 0 1 46.51 10.3" />
            </g>
        </svg>
    </div>
</div>


<h1 class="dbtl-header">Dry Lab</h1>


<div class="cycle-card">
    <div class="cycle-left">
        <div>
            <h2>Cycle 1</h2>
            <div class="phase-tabs">
                <div class="phase-tab active" data-tab="design5">Design</div>
                <div class="phase-tab" data-tab="build5">Build</div>
                <div class="phase-tab" data-tab="test5">Test</div>
                <div class="phase-tab" data-tab="learn5">Learn</div>
            </div>
        </div>
    </div>
    <div class="cycle-right">
        <div class="phase-content active" id="design5">
            <div class="phase-title">Design</div>
            <div class="phase-subtitle">Large-scale model of eutrophication:</div>
            <p>Based on our initial ideas, we had the idea to model on a large scale how eutrophication affects water bodies, using test cases of some of the rivers across the world that have been most hard hit by eutrophication due to nitrate leaching gone amok, and how our bacteria would curb and control these effects</p>
            <br>
            <p>
                We aimed to create an interactive UI that would depict how the selected water body would change over the years, with and without our project.
            </p>

            <div class="phase-subtitle">Biofilm modeling:</div>
            <p>
                To study the microenvironment created within the biofilms, as one of our main concerns was that <em>P. putida</em> Y-9 is strictly aerobic, we sought to model the oxygen concentration of oxygen to see whether micro-oxic environments would be created, which would pose a huge problem to us!
            </p>

            <div class="phase-subtitle">GEM models </div>
            <p>
                We came across GEM models, which are networks of all the metabolites, enzymes, and corresponding genes involved, and tried to use them as a base for modeling the reactions and kinetics of metabolite concentrations. 
            </p>
        </div>

        <div class="phase-content" id="build5">
            <div class="phase-title">Build</div>
            <p>Once we had figured out our ideas, we sought out to look for studies and literature on eutrophication, especially in the Indian subcontinent.</p>
            <br>
            <p>We looked into the literature of biofilm formation. Already existing tools and models which we could use.</p>
            <br>
            <p>We looked for GEMs of the <em>P. putida</em> bacteria family which may perform DNRA all in hopes of better understanding the metabolic activities of the organism in context to DNRA specifically. This also included learning how to work with the cobra.py library which is used to work with GEMs.</p>
        </div>


        <div class="phase-content" id="test5">
            <div class="phase-title">Test</div>
            <p>Over time, <em>P. putida</em> Y-9 seemed to lead us to dead ends. With not many resources that would verify the claims of the original papers and no responses from the authors of the papers, we decided to shift our focus to a different microbe. We were not even able to find any genomic data of this strain, nor the amino acid sequence of the relevant proteins.
</p>
<br>
<p>
           We kept testing and playing with GEMs of organisms that perform DNRA, varying the nitrate and ammonium content as input flux and inhibiting specific genes to see how it would affect bacterial growth and the output flux of ammonium.
</p>
        </div>
        <div class="phase-content" id="learn5">
            <div class="phase-title">Learn</div>
            <p>While trying to model just the large-scale eutrophication of a lake over years, we realised that this would be far too complex a task for us to reasonably implement with our skills and the time we had, as we had to account for the complex interactions between agricultural soil, runoff water, groundwater, and rivers.</p>
            <br>
            <p>While moving through the project, this model becomes redundant, as by switching strains to <em>B. salipalidus</em>, these species can perform DNRA regardless of oxygen concentration.</p>
            <br>
            <div class="callout-box"><p>While looking more into GEM models, and upon the inputs of Dr. Nagasuma Chandra, we realised that this line of work was irrelevant to our project, as GEM models provided no information on kinetics.</p></div>
        </div>
    </div>

    <div class="plasmid">
        <svg viewBox="0 0 100 100">
            <defs>
                <marker id="arrowhead" markerWidth="2.5" markerHeight="2.5" refX="1.25" refY="1.25" orient="auto">
                    <polygon points="0 0, 2.5 1.25, 0 2.5" fill="var(--phase-build)" />
                </marker>
            </defs>
        
            <circle class="plasmid-bg" cx="50" cy="50" r="40"/>

            <g transform="rotate(-90 50 50)">
                <path id="segment-1" class="plasmid-segment" d="M 53.49 10.3 A 40 40 0 0 1 89.7 46.51" />
                <path id="segment-2" class="plasmid-segment" d="M 90 53.49 A 40 40 0 0 1 53.49 89.7" marker-end="url(#arrowhead)" />
                <path id="segment-3" class="plasmid-segment" d="M 13.62 66.86 A 40 40 0 0 0 33.14 86.38" />
                <path id="segment-4" class="plasmid-segment" d="M 10 46.51 A 40 40 0 0 1 46.51 10.3" />
            </g>
        </svg>
    </div>
</div>

<div class="cycle-card">
    <div class="cycle-left">
        <div>
            <h2>Cycle 2</h2>
            <div class="phase-tabs">
                <div class="phase-tab active" data-tab="design2">Design</div>
                <div class="phase-tab" data-tab="build2">Build</div>
                <div class="phase-tab" data-tab="test2">Test</div>
                <div class="phase-tab" data-tab="learn2">Learn</div>
            </div>
        </div>
    </div>
    <div class="cycle-right">
        <div class="phase-content active" id="design2">
            <div class="phase-title">Design</div>
            <div class="phase-subtitle">Soil modeling</div>
            <p>After realizing the complexity of our goals from the first cycle, we decided to model soil dynamics around plant roots. We took into account the nutrient uptake rate of the plants, nitrite and nitrate leaching in vertical and horizontal directions, and variation with depth.</p>

            <div class="phase-subtitle">Pathway modelling</div>
            <p>As we went through the literature on DNRA and ANRA pathways, we realised they are very poorly characterized. We were unable to find any kinetic characterization of the pathways at an enzymatic level, or the regulations on the production of involved genes either. So, we collected data on the rate of DNRA in different pH, nitrate, and ammonium concentrations in hopes of finding some expression for <em>nar</em> and <em>nir</em> enzyme expression.</p>

            <div class="phase-subtitle">Correlating various soil factors</div>
            <p>To fit our model to real-world conditions, we needed to find how various soil factors like pH, nitrate and ammonium concentration, temperature, rainfall, latitude, and soil type relate with each other.</p>

            <div class="phase-subtitle">Genome comparison</div>
            <p>To identify the relevant proteins in the newly discovered <em>Neobacillus</em> sp., we BLASTed the genomes of these three bacteria against other well-known and well-characterized species like <em>Bacillus subtilis</em>, to identify the enzymes and proteins relevant to biofilm formation. We further planned on using PGAP (Prokaryotic Genome Annotation Pipeline), a genome annotation tool by NCBI, to identify any relevant enzymes present in newly isolated soil bacteria from agricultural soil.</p>
            
            <div class="phase-subtitle">Protein modeling</div>
            <p>We decided to model the different proteins obtained from using different restriction enzymes, since they added an extra terminal sequence of amino acids to the protein.</p>
            <div class="callout-box">
            <p>With the guidance of Dr. Mahavir, we decided to model the different proteins obtained from using different restriction enzymes, since they added an extra terminal sequence of amino acids to the protein. We did this to help us choose a restriction enzyme that would not interfere with enzyme structure and functionality. We modeled three structures of the proteins with all the various changes and compared the structure on AlphaFold.</p>
            </div>
        </div>
        <div class="phase-content" id="build2">
            <div class="phase-title">Build</div>
            <p>Upon trying to model the soil conditions, we once again realised that the complexity of this endeavor was very high, so we decided to abandon this.</p>
            <div class="callout-box">
                <p>While trying to characterise the kinetics and protein expression, at the guidance of Mr. Aditya, we realised due to several reasons that this would be much harder than anticipated, since we didn’t have the capabilities to isolate and study the enzymes and their expression in the lab. Furthermore, the discrepancy between the order of magnitudes of enzyme production and rate of change of metabolite concentration would pose more problems.</p>
            </div>
            <p>The BLAST comparisons of the Neobacillus species were widely successful, and we were able to successfully identify the exact subunits involved in the operons using the <em>B. subtilis</em> genome.</p>
        </div>
        <div class="phase-content" id="test2">
            <div class="phase-title">Test</div>
            <p>These subunits had not been characterized by the research group that identified the species, but we identified the orthologous genes with very high accuracy. Upon confirmation with literature, we were able to confirm that the binding sites and active regions matched exactly with the prediction.<br>We also identified the presence of certain genes that are crucial in biofilm formation.</p>
            <p>We modeled the different structures of proteins arising from the various choices of restriction enzymes on AlphaFold, and upon analysis and comparison, we realised that for our choice of restriction sites, there was negligible change in protein structure and functionality.</p>
        </div>
        <div class="phase-content" id="learn2">
            <div class="phase-title">Learn</div>
            <p>From our soil modelling attempts, we learned how the complexity of accurately simulating soil dynamics with root interactions and leaching was beyond our current scope and resources.</p>
            <p>The pathway modelling attempts made us realize the pursuit is futile, given we had little existing enzymatic data and lacked proper resources to conduct our own experiments in that regard.</p>
            <p>Through genome comparison tests, we could identify orthologous genes, confirm the active sites with literature, and discover uncharacterized biofilm-related genes.</p>
            <p>From protein modeling results, we learned that the terminal amino acid additions from the chosen restriction enzymes did not affect the protein structure significantly. Thus, we decided to go forth with the most economical and easily available restriction enzyme for the wet lab.</p>
        </div>
    </div>

    <div class="plasmid">
        <svg viewBox="0 0 100 100">
            <defs>
                <marker id="arrowhead" markerWidth="2.5" markerHeight="2.5" refX="1.25" refY="1.25" orient="auto">
                    <polygon points="0 0, 2.5 1.25, 0 2.5" fill="var(--phase-build)" />
                </marker>
            </defs>
        
            <circle class="plasmid-bg" cx="50" cy="50" r="40"/>

            <g transform="rotate(-90 50 50)">
                <path id="segment-1" class="plasmid-segment" d="M 53.49 10.3 A 40 40 0 0 1 89.7 46.51" />
                <path id="segment-2" class="plasmid-segment" d="M 90 53.49 A 40 40 0 0 1 53.49 89.7" marker-end="url(#arrowhead)" />
                <path id="segment-3" class="plasmid-segment" d="M 13.62 66.86 A 40 40 0 0 0 33.14 86.38" />
                <path id="segment-4" class="plasmid-segment" d="M 10 46.51 A 40 40 0 0 1 46.51 10.3" />
            </g>
        </svg>
    </div>
</div>

<div class="cycle-card">
    <div class="cycle-left">
        <div>
            <h2>Cycle 3</h2>
            <div class="phase-tabs">
                <div class="phase-tab active" data-tab="design3">Design</div>
                <div class="phase-tab" data-tab="build3">Build</div>
                <div class="phase-tab" data-tab="test3">Test</div>
                <div class="phase-tab" data-tab="learn3">Learn</div>
            </div>
        </div>
    </div>
    <div class="cycle-right">
        <div class="phase-content active" id="design3">
            <div class="phase-title">Design</div>
            <div class="phase-subtitle">Modelling genetically modified system</div>
            <p>Upon realising the complexity, we decided to model the pathway in the bacterium where the enzymes are being expressed at a constant rate by a constitutive promoter. Another factor that pushed us to pursue modelling of the final bacterium over the chassis was the synergy it provides to the rest of our project, as doing this would allow us to predict the conversion rate under various conditions, according to which it can be applied.</p>

            <div class="phase-subtitle">Fixing GenBank UI</div>
            <p>After using the GenBank framework extensively for BLAST comparisons, we realised that the tool is highly inefficient and very inconvenient to use, especially for someone from a traditional biology background with no experience. Hence, we decided to try to improve this framework to improve the ease for future iGEM teams.</p>

            <div class="phase-subtitle">Using ML models to find correlations</div>
            <p>Upon realising the huge number of variables involved in soil conditions, we decided to use an ML approach to train on large data sets of soil conditions, based on which it would predict desired output.</p>

            <div class="phase-subtitle">Product scaling</div>
            <p>Using the model to calculate the nitrate to ammonium conversion rates, we planned to use this to guide the commercial application of our GMO bacterium into soil, by fine-tuning the biomass of the bacterium and the amount of fertilizer that needs to be applied for certain soil types and plant species for optimal growth and minimal nitrogen loss.</p>
        </div>

        <div class="phase-content" id="build3">
            <div class="phase-title">Build</div>
            <p>We began developing the model of the genetically modified bacterium, incorporating constant enzyme expression through a constitutive promoter. This helped us simplify the system and make it more tractable.</p>
            <p>We started planning an improved version of the GenBank framework with a more user-friendly interface to assist future users, especially those without strong computational backgrounds.</p>
            <p>For the ML model, we began identifying publicly available soil datasets that included variables such as pH, temperature, rainfall, and nitrogen content to serve as training data.</p>
            <p>In preparation for scaling, we conceptualized how the predictive model could be integrated with real-world soil data to generate fertilizer optimization recommendations.</p>
        </div>

        <div class="phase-content" id="test3">
            <div class="phase-title">Test</div>
            <p>Initial testing focused on simulating the metabolic pathway of the modified bacterium to verify that constant enzyme expression provided stable conversion rate predictions under different conditions.</p>
            <p>We benchmarked the GenBank system to identify the most critical UI components needing improvement. We decided to focus on the BLAST framework.</p>
            <p>Preliminary ML runs were planned to evaluate the model’s ability to correlate key soil parameters and predict nitrate-to-ammonium ratios.</p>
            <img src="https://static.igem.wiki/teams/6006/wiki/misc/unnamed3.avif" alt="MSE loss vs Epoch" class="animate-on-scroll" style="display: block; margin: 1rem auto 0; max-width: 800px; border-radius: 1rem; box-shadow: 0 10px 15px -3px rgba(0,0,0,0.1); margin-bottom: 2rem;">
            <p>The framework for product scaling was tested conceptually by assessing if model outputs could meaningfully inform bacterial biomass and fertilizer input ratios for different soil types.</p>
        </div>

        <div class="phase-content" id="learn3">
            <div class="phase-title">Learn</div>
            <p>From the theory on genetically modified bacterium circuits, we learnt that simplifying enzyme expression to a constant rate made pathway simulations much more manageable while maintaining biological relevance.</p>
            <p>From evaluating the GenBank framework, we confirmed that improving its interface could greatly enhance usability for biologists without computational training.</p>
            <p>Upon inputs from our senior Armaan Khetarpaul, we decided to use XGBoost models instead of neural networks, since they are less prone to overfitting.</p>
            <p>From the product scaling concept, we learnt that the combination of pathway modelling and soil condition predictions could directly guide practical applications, improving both efficiency and sustainability.</p>
        </div>
    </div>

    <div class="plasmid">
        <svg viewBox="0 0 100 100">
            <defs>
                <marker id="arrowhead" markerWidth="2.5" markerHeight="2.5" refX="1.25" refY="1.25" orient="auto">
                    <polygon points="0 0, 2.5 1.25, 0 2.5" fill="var(--phase-build)" />
                </marker>
            </defs>
        
            <circle class="plasmid-bg" cx="50" cy="50" r="40"/>

            <g transform="rotate(-90 50 50)">
                <path id="segment-1" class="plasmid-segment" d="M 53.49 10.3 A 40 40 0 0 1 89.7 46.51" />
                <path id="segment-2" class="plasmid-segment" d="M 90 53.49 A 40 40 0 0 1 53.49 89.7" marker-end="url(#arrowhead)" />
                <path id="segment-3" class="plasmid-segment" d="M 13.62 66.86 A 40 40 0 0 0 33.14 86.38" />
                <path id="segment-4" class="plasmid-segment" d="M 10 46.51 A 40 40 0 0 1 46.51 10.3" />
            </g>
        </svg>
    </div>
</div>

<div class="cycle-card">
    <div class="cycle-left">
        <div>
            <h2>Cycle 4</h2>
            <div class="phase-tabs">
                <div class="phase-tab active" data-tab="design4">Design</div>
                <div class="phase-tab" data-tab="build4">Build</div>
                <div class="phase-tab" data-tab="test4">Test</div>
                <div class="phase-tab" data-tab="learn4">Learn</div>
            </div>
        </div>
    </div>
    <div class="cycle-right">
        <div class="phase-content active" id="design4">
            <div class="phase-title">Design</div>
            <div class="phase-subtitle">Soil ML</div>
            <p>We started building the XGBoost model, and applied a log(1+x) transformation, which stabilizes variance and improves model learning on our ammonium data, which was positively skewed data. Simultaneously, once again at the suggestion of our senior Armaan Khetarpaul, we decided to use synthetic data to create data sets for the ammonium concentration in soil after our bacterium is applied.</p>

            <div class="phase-subtitle">ODE model</div>
            <p>We decided to build the model similar to the previous iGEM teams, Cattlelyst and Halocleen, using a system of ODEs, characterising the concentrations of metabolites inside the cell and in the surrounding medium. We accounted for transport reactions between these two compartments and assimilation of intracellular ammonium by the GS/GOGAT cycles. Later on, we also incorporated the leaching of extracellular nitrate and the uptake of extracellular ammonium and nitrate by plants.</p>

            <div class="phase-subtitle">Genbank UI</div>
            <p>Our goal was to modernize and simplify working with NCBI’s GenBank. Addressing issues like poor search accuracy, lack of context explanations, and weak visualization options. We created a list of pre-existing biotech tools online and consulted bioengineering enthusiasts and faculty about problems with these resources.</p>
        </div>

        <div class="phase-content" id="build4">
            <div class="phase-title">Build</div>
            <div class="phase-subtitle">Soil ML</div>
            <p>We used an XGBoost Regressor with hyperparameters tuned through a previous search, with our chosen configuration emphasizing balanced learning with moderate regularization. We applied a smearing estimate (a standard bias correction technique) when converting predictions back to the original scale. To generate the synthetic data, we used our ODE model and literature to predict by what percentage the rate of DNRA will increase in the soil.</p>
            
            <div class="phase-subtitle">ODE model</div>
            <p>We built the model on COPASI software, drawing inspiration from the previous iGEM team Cattlelyst from Wageningen 2021, to construct the ODEs and obtain some of the rate constants. We obtained the other values from previous characterisations of the enzymes from the literature. We later expanded the model to include nitrate leaching and N uptake by plants at a constant rate.</p>

            <div class="phase-subtitle">Genbank UI</div>
            <p>We created a preliminary framework in Python, learnt to bypass the use of Pio-python.</p>
        </div>

        <div class="phase-content" id="test4">
            <div class="phase-title">Test</div>
            <div class="phase-subtitle">Soil ML</div>
            <p>We evaluated the model on the test set using standard regression metrics on the original ammonium scale. The results obtained were pretty good considering the small data size, explaining almost 60% of the variation in ammonium levels, meaning that the model is picking up on many important patterns.</p>
            <p>After going through the literature and simulating conditions on our ODE model, and running the ML model on some different conditions, we finally settled on a percentage increase value obtained by comparing the maximum conversion rates in our model and some rates from the literature and the previous iGEM team Wageningen.</p>

            <div class="phase-subtitle">ODE model</div>
            <p>We had several testing phases by changing constant values and observing conversion rates, and noticed that there were two issues initially: unrealistically high cell growth and accumulation of ammonium in the cellular component. We fixed these issues by capping the cell growth at a fixed value and by restricting ammonium import beyond a certain concentration of cellular ammonium.</p>
            <p>We also varied the enzyme concentration across various values and observed the cellular nitrate values, which were all within permissible values and would not cause toxicity. While varying the rate of enzyme concentration, we also observed a wide range of conversion efficiencies from about 10% to 98%, and at very high concentrations (>M) of enzyme, the model broke down.</p>
            <p>Upon including the leaching and uptake, and varying the values and observing efficiency,y and by comparing with real-world values, we decided that the most sensible approximation for the rate would be 10<sup>-11</sup>mol/L. We also switched around enzyme concentrations after this inclusion and observed that in some ranges the change in efficiency is drastic, but in others there was barely any change.</p>

            <div class="phase-subtitle">Genbank UI</div>
            <p>For testing, we presented our solutions to said-tool users for their feedback and insights on how to improve on it. We optimized the code in JavaScript and built a GitHub website to host our tool.</p>
        </div>

        <div class="phase-content" id="learn4">
            <div class="phase-title">Learn</div>
            <div class="phase-subtitle">Soil ML</div>
            <p>We realised that the dataset being quite small, with only 902 samples, likely limits the model’s performance. More data would help it learn more nuanced patterns and improve accuracy further. We believe that future iGEM teams should focus on expanding the dataset with more diverse soil samples and conditions. This would allow for training more robust models that can better capture the complexity of soil ammonium dynamics.</p>
            
            <div class="phase-subtitle">ODE model</div>
            <p>From the huge amount of testing done, we figured out the limits of the model and what are the realistic ranges of enzyme concentration, cell volume, uptake, and leaching value, which are vital for being able to predict optimal conditions for given conditions of soil and plant type. This testing helped us understand the potential utility of our model in designing Wet Lab experiments, by constraining certain factors such as the ratio of nitrate: nitrite reductase to ensure no nitrite build-up. It also helps us better characterize the functionality of the modified bacterium so that we at least know a ballpark estimate of what the expected results should be.</p>
            
            <div class="phase-subtitle">Genbank UI</div>
            <p>We hope the tool we developed can be of use to people in the field in some way. We were able to learn a lot of new things about how things actually work behind the simple webpage that we see. But most importantly, the development of GenBankUI was a fun way to interact with our juniors, who helped us with the development of this tool! We could familiarize them with the world of synbio and iGEM.</p>
        </div>
    </div>

    <div class="plasmid">
        <svg viewBox="0 0 100 100">
            <defs>
                <marker id="arrowhead" markerWidth="2.5" markerHeight="2.5" refX="1.25" refY="1.25" orient="auto">
                    <polygon points="0 0, 2.5 1.25, 0 2.5" fill="var(--phase-build)" />
                </marker>
            </defs>
        
            <circle class="plasmid-bg" cx="50" cy="50" r="40"/>

            <g transform="rotate(-90 50 50)">
                <path id="segment-1" class="plasmid-segment" d="M 53.49 10.3 A 40 40 0 0 1 89.7 46.51" />
                <path id="segment-2" class="plasmid-segment" d="M 90 53.49 A 40 40 0 0 1 53.49 89.7" marker-end="url(#arrowhead)" />
                <path id="segment-3" class="plasmid-segment" d="M 13.62 66.86 A 40 40 0 0 0 33.14 86.38" />
                <path id="segment-4" class="plasmid-segment" d="M 10 46.51 A 40 40 0 0 1 46.51 10.3" />
            </g>
        </svg>
    </div>
</div>

</div>


{% endblock %}


{% block scripts %}
<script src="{{ url_for('static', filename='engineering.js') }}"></script>
{% endblock %}